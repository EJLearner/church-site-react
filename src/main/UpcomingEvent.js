import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './UpcomingEvent.css';

UpcomingEvent.propTypes = {
  date: PropTypes.string.isRequired,
  lines: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

function UpcomingEvent(props) {
  function renderLines(lines) {
    return lines.map((line, index) => (
      <div className="other-line" key={index}>
        {line}
      </div>
    ));
  }

  return (
    <div className="event-area">
      <div className="event-date">
        {moment(props.date).format('MMM.')}
        <br />
        {moment(props.date).format('D')}
      </div>
      <div className="event-title">{props.title}</div>
      <div className="other-lines">{renderLines(props.lines)}</div>
    </div>
  );
}

export default UpcomingEvent;
