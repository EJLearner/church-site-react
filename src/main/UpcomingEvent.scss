@import '../variables';

.event-area {
  text-align: center;
  width: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 25px 0 25px;
  flex: 1 0 auto;
}

.event-date {
  border: 1px solid black;
  border-radius: 50%;
  width: 80px;
  height: 80px;
  text-align: center;
}

.event-title {
  color: $red;
}

.event-title,
.event-line {
  line-height: 200%;
}
