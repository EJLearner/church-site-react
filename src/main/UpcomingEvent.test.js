import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import UpcomingEvent from './UpcomingEvent';

describe('renders', () => {
  let props;

  beforeEach(() => {
    props = {
      date: '2011-05-10',
      title: 'test title',
      lines: ['line 1', 'line 2']
    };
  });

  it('does not crash', () => {
    expect(() => shallow(<UpcomingEvent {...props} />)).not.toThrow();
  });

  it('renders correctly when lines are strings', () => {
    const snap = renderer.create(<UpcomingEvent {...props} />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('renders correctly when lines are react nodes', () => {
    props.lines = [<span key="1">Line 1T</span>, <span key="2">Line 2T</span>];
    const snap = renderer.create(<UpcomingEvent {...props} />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
