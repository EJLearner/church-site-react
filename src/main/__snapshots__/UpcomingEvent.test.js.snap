// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders correctly when lines are react nodes 1`] = `
<div
  className="event-area"
>
  <div
    className="event-date"
  >
    May.
    <br />
    10
  </div>
  <div
    className="event-title"
  >
    test title
  </div>
  <div
    className="other-lines"
  >
    <div
      className="other-line"
    >
      <span>
        Line 1T
      </span>
    </div>
    <div
      className="other-line"
    >
      <span>
        Line 2T
      </span>
    </div>
  </div>
</div>
`;

exports[`renders renders correctly when lines are strings 1`] = `
<div
  className="event-area"
>
  <div
    className="event-date"
  >
    May.
    <br />
    10
  </div>
  <div
    className="event-title"
  >
    test title
  </div>
  <div
    className="other-lines"
  >
    <div
      className="other-line"
    >
      line 1
    </div>
    <div
      className="other-line"
    >
      line 2
    </div>
  </div>
</div>
`;
